src/dune_rules/gen_rules.ml

```diff
@@ -112,12 +112,13 @@ end = struct
       let+ () = toplevel_setup ~sctx ~dir ~toplevel in
       empty_none
     | Library.T lib ->
-      (* XXX why are we setting up private doc rules for disabled libraries? *)
-      let* () = Odoc.setup_private_library_doc_alias sctx ~scope ~dir:ctx_dir lib
-      and+ enabled_if = Lib.DB.available (Scope.libs scope) (Library.best_name lib) in
+      let* enabled_if = Lib.DB.available (Scope.libs scope) (Library.best_name lib) in
       if_available_buildable
         ~loc:lib.buildable.loc
-        (fun () -> Lib_rules.rules lib ~sctx ~dir ~scope ~dir_contents ~expander)
+        (fun () ->
+          let+ () = Odoc.setup_private_library_doc_alias sctx ~scope ~dir:ctx_dir lib
+          and+ rules = Lib_rules.rules lib ~sctx ~dir ~scope ~dir_contents ~expander in
```

Comment:
style question for @rgrinberg: is there an advantage in doing this compared to the following (I've noticed that pattern in odoc rules in particular)

```ocaml
(fun () ->
  let* () = Odoc.setup_private_library_doc_alias sctx ~scope ~dir:ctx_dir lib in
  Lib_rules.rules lib ~sctx ~dir ~scope ~dir_contents ~expander)
```

Reply:
The former has the opportunity to do something concurrently while the the latter does not. Usually it doesn't matter when generating rules as we rarely do work where we can take advantage of it. However, as we make dune more concurrent and allow for more dynamism (e.g. dune file generation), the latter becomes more important.